crates/turborepo-lib/src/config/client.rs

```diff
@@ -60,17 +62,12 @@ impl ClientConfigLoader {
             .add_source(Environment::with_prefix("turbo").source(environment))
             .set_override_option("remote_cache_timeout", remote_cache_timeout)?
             .build()?
+            // Deserialize is the only user-input-fallible step.
+            // Everything else is programmer error.
+            // This goes wrong when TURBO_REMOTE_CACHE_TIMEOUT can't be deserialized to u64
             .try_deserialize();
 
-        // This goes wrong when TURBO_REMOTE_CACHE_TIMEOUT can't be deserialized to u64
-        match config_attempt {
-            Err(_) => Ok(ClientConfig {
-                config: ClientConfigValue {
-                    remote_cache_timeout: DEFAULT_TIMEOUT,
-                },
-            }),
-            Ok(config) => Ok(ClientConfig { config }),
-        }
```

Comment:
This behavior does not match any of our other configuration loaders which fail hard in the event of garbage input.

We can come back and create "this error is only a warning" behavior if we want to do so later, but for now make the failure mode consistent.